### ISSUE TITLE
Build instructions

### ISSUE THREAD

**Issue by yairchu** [cdfa/frugel#1]
I saw that there is a `stack.yaml` file so I tried `stack build`, but it failed:

```
% stack build

Error: While constructing the build plan, the following exceptions were encountered:

In the dependencies for frugel-0.0.2.0:
    Interpolation needed, but the stack configuration has no specified version  (latest matching version is 0.3.0)
    QuickCheck needed, but the stack configuration has no specified version  (latest matching version is 2.14.2)
    aeson needed, but the stack configuration has no specified version  (latest matching version is 2.0.2.0)
    base-noprelude must match >=4.7 && <5, but the stack configuration has no specified version  (latest matching
                   version is 4.13.0.0)
    composition needed, but the stack configuration has no specified version  (latest matching version is 1.0.2.2)
    dictionary-sharing needed, but the stack configuration has no specified version  (latest matching version
                       is 0.1.0.0)
    generic-data must match >=0.8.0.0 && <0.9, but the stack configuration has no specified version  (latest matching
                 version is 0.8.3.0)
    genvalidity needed, but the stack configuration has no specified version  (latest matching version is 1.0.0.0)
    genvalidity-containers needed, but the stack configuration has no specified version  (latest matching version
                           is 1.0.0.0)
    genvalidity-text needed, but the stack configuration has no specified version  (latest matching version
                     is 1.0.0.0)
    indexed-profunctors needed, but the stack configuration has no specified version  (latest matching version
                        is 0.1.1)
    it-has needed, but the stack configuration has no specified version  (latest matching version is 0.2.0.0)
    jsaddle must match >=0.9.8 && <0.10, but the stack configuration has no specified version  (latest matching
            version is 0.9.8.0)
    jsaddle-warp must match ==0.9.*, but the stack configuration has no specified version  (latest matching version
                 is 0.9.8.0)
    lens needed, but the stack configuration has no specified version  (latest matching version is 5.1)
    megaparsec must match >=9.0.1 && <9.1, but the stack configuration has no specified version  (latest matching
               version is 9.0.1)
    miso must match >=1.8.0.0 && <1.9, but the stack configuration has no specified version  (latest matching version
         is 1.8.0.0)
    multiset needed, but the stack configuration has no specified version  (latest matching version is 0.3.4.3)
    optics must match ==0.4.*, but the stack configuration has no specified version  (latest matching version is 0.4)
    optics-core must match ==0.4.*, but the stack configuration has no specified version  (latest matching version
                is 0.4)
    optics-vl must match >=0.2.1 && <0.3, but the stack configuration has no specified version  (latest matching
              version is 0.2.1)
    parser-combinators needed, but the stack configuration has no specified version  (latest matching version
                       is 1.3.0)
    pretty-show needed, but the stack configuration has no specified version  (latest matching version is 1.10)
    prettyprinter must match >=1.7.0 && <1.8, but the stack configuration has no specified version  (latest matching
                  version is 1.7.1)
    profunctors needed, but the stack configuration has no specified version  (latest matching version is 5.6.2)
    relude must match >=1.0.0.0, but the stack configuration has no specified version  (latest matching version
           is 1.0.0.1)
    size-based needed, but the stack configuration has no specified version  (latest matching version is 0.1.2.0)
    testing-feat needed, but the stack configuration has no specified version  (latest matching version is 1.1.0.0)
    validity-containers needed, but the stack configuration has no specified version  (latest matching version
                        is 0.5.0.4)
    validity-text needed, but the stack configuration has no specified version  (latest matching version is 0.3.1.2)
    wai-app-static must match >=3.1.7 && <3.2, but the stack configuration has no specified version  (latest matching
                   version is 3.1.7.2)
    warp must match >=3.3.17 && <3.4, but the stack configuration has no specified version  (latest matching version
         is 3.3.18)
    websockets must match >=0.12.7 && <0.13, but the stack configuration has no specified version  (latest matching
               version is 0.12.7.3)
needed since frugel is a build target.

Some different approaches to resolving this:

  * Recommended action: try adding the following to your extra-deps in /Users/yairchu/dev/src/frugel/stack.yaml:

- Interpolation-0.3.0@sha256:b0381214e6036ed2b6a0a04b7e4ec89736ca531ef03e315a3ec2c494059ff8fb,1319
- QuickCheck-2.14.2@sha256:4ce29211223d5e6620ebceba34a3ca9ccf1c10c0cf387d48aea45599222ee5aa,7736
- aeson-2.0.2.0@sha256:5720fffb7289366029f2b7940e9f8b22a1b4c282f0cef4710685b1d14d76bdc7,6327
- base-noprelude-4.13.0.0@sha256:3cccbfda38e1422ca5cc436d58858ba51ff9114d2ed87915a6569be11e4e5a90,6842
- composition-1.0.2.2@sha256:fab25a9386057cdd83094109ac204c8aa7404e19fdfec6708ad383cd3a2d6fe5,659
- dictionary-sharing-0.1.0.0@sha256:6990c4cb6e5c7fd2b3c545576accc2c00a56cdc52ffe646f70680d781893c7d6,788
- generic-data-0.8.3.0@sha256:fef21320697370fa1fadc14f72ac3ad8092b021f5371b315be3c9f3bde44c274,3710
- genvalidity-1.0.0.0@sha256:3fc1b2069ed528e52338ad12a6ee41079bc151f22cc3dc33b5e317c4cd0046b5,2758
- genvalidity-containers-1.0.0.0@sha256:ff0948d748187a2aff2158a3aca8d5245813074deb6209ff44d3e80d38435c55,2230
- genvalidity-text-1.0.0.0@sha256:75e7c5dbb850598f627572698c06f795e896208b082c2c3070301ebf748654a7,1832
- indexed-profunctors-0.1.1@sha256:363e1bcd89f8498a770ae783e8388fcce35e6515b4c38dca47327aad81196a7b,1063
- it-has-0.2.0.0@sha256:e0dfa84dfba835d8616b817305a80878383b8ba7ebe3c9802f33b312e7110d76,1409
- jsaddle-0.9.8.0@sha256:d7acf2247cc93c2d8d0959eccf627077c4175c81b762c8d107c7006dda662d05,4418
- jsaddle-warp-0.9.8.0@sha256:d1c75ae6ab519f923bc8a13775abd779e584bd17c96a49a76f799860bd9174e9,2713
- lens-5.1@sha256:eb01fc4b1cfbad0e94c497eaf7b9f0e9b6c3dc7645c8b4597da7dc9d579f8500,14519
- megaparsec-9.0.1@sha256:2244ab77521a52a0083966c5405b3151f149e1db50b1a5b104cd9b1e56617203,3248
- miso-1.8.0.0@sha256:090e8702d6ffd096e79ea083f5f3293737e3ec44a68b21b8013dc445b92064f0,4339
- multiset-0.3.4.3@sha256:968192524bd38ffa29856e1e3f9f6f8a5704f98899d4b25f8ebd425001594f61,1790
- optics-0.4@sha256:9fb69bf0195b8d8f1f8cd0098000946868b8a3c3ffb51e5b64f79fc600c3eb4c,6568
- optics-core-0.4@sha256:59e04aebca536bd011ae50c781937f45af4c1456af1eb9fb578f9a69eee293cd,4995
- optics-vl-0.2.1@sha256:ccaf2713358c6940294b0979df0246af9a03c1410cd559c0a86f68d9c874462d,1406
- parser-combinators-1.3.0@sha256:edd54ba56cbae8fadbcceebcfef31b2c70a835e92e5eda41151b939c40647281,1570
- pretty-show-1.10@sha256:bfef8da0ac4db4fc7f41edfec9e9a701ac206418d70702bff1e40f20d6a2b1f1,1888
- prettyprinter-1.7.1@sha256:9c43c9d8c3cd9f445596e5a2379574bba87f935a4d1fa41b5407ee3cf4edc743,6987
- profunctors-5.6.2@sha256:3d3685119243a7ebf984fa6af03299d156ab7674a432e2e15ecee2a4fd420fb6,2483
- relude-1.0.0.1@sha256:35bcdaf14018e79f11e712b0e2314c1aac79976f28f4adc179985457493557d5,11569
- size-based-0.1.2.0@sha256:1b33da89d270189661dbbda49a88b0c21d2fc2f7a407e7a2b1933e2faf0f5d4d,1258
- testing-feat-1.1.0.0@sha256:7c7629c5014edf06aefbf30a061d1ee64c6ee15f438d868e34749fb22208ab0b,2466
- validity-containers-0.5.0.4@sha256:b7ae4711646de5f44ef92ce6ae01836bab219ac4f8ebd9a8e05dd073eb0c1d99,1154
- validity-text-0.3.1.2@sha256:37b91c473c148de41703d4a4bfdd9067924dde1ae7258c0d012dcf0404517fff,910
- wai-app-static-3.1.7.2@sha256:ad6b8b07777e6d63f5bf84da2522ac469ff66219a59cdb72baeb69af95e4ffe0,4170
- warp-3.3.18@sha256:e583b66084c7ecfd3f59349aeae72523b90e7887d3523868c6f22fe719172054,10910
- websockets-0.12.7.3@sha256:c4133af18fb420527b30507a69ff20d6819b79f02643051f1ed97aba0468846c,7976

Plan construction failed.
```

**Comment by cdfa** [comment]
Hey @yairchu,  nice to see you here. Apologies for the lack of documentation, I've been very busy finishing up the thesis. I will add proper build instructions soon, but building with cabal should work (at least it did for me on Windows (using https://community.chocolatey.org/packages/haskell-dev)).
The build system I primarily used is Nix. This is required for building the web version and may provide more reproducibility than cabal. `shell.nix` also defines an extensive development environment.

**Comment by cdfa** [comment]
Fixed in 260874274fcc2ad0dfc21dd644637c5d8bd62545 !
